File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.java
class Solution {
    public int[][] onesMinusZeros(int[][] grid) {
        int m = grid.length, n = grid[0].length;
        int[] rowOnes = new int[m], colOnes = new int[n];
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                if (grid[i][j] == 1) {
                    rowOnes[i]++;
                    colOnes[j]++;
                }
        int[][] res = new int[m][n];
        for (int i = 0; i < m; i++) {
            int rowTerm = 2 * rowOnes[i] - n;
            for (int j = 0; j < n; j++)
                res[i][j] = rowTerm + 2 * colOnes[j] - m;
        }
        return res;
    }
}

File: 2482-difference-between-ones-and-zeros-in-row-and-column/2482-difference-between-ones-and-zeros-in-row-and-column.cpp
class Solution {
public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();

        vector<int> onesRow(m, 0), onesCol(n, 0);
        vector<vector<int>> res(m, vector<int>(n, 0));

       
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                if (grid[i][j] == 1) {
                    onesRow[i]++;
                    onesCol[j]++;
                }
            }
        }

       
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                res[i][j] = 2 * onesRow[i] + 2 * onesCol[j] - m - n;
            }
        }

        return res;
    }
};
